### ISSUE TITLE
Online EWC fails if multi-head model has different number of classes for each head

### ISSUE THREAD

**Issue by SunHaozhe** [ContinualAI/avalanche#954]
The current implementation of Online EWC does not work when the multi-head model has different number of classes for each classification head.

**Comment by AntonioCarta** [comment]
Thanks! We will try to look into this ASAP.

**Comment by AndreaCossu** [comment]
I am not able to reproduce this issue, it's working for me on Split MNIST (4 experiences, 3-2-3-2 classes in each exp) with multi-head MLP. 
Can you provide a minimal working code? 
Also, are you using the master version of avalanche or the beta release from pip? I think @AntonioCarta solved this bug in a recent commit which is not included in the beta version.

**Comment by AndreaCossu** [comment]
Btw this is my code:
```python
import torch
from os.path import expanduser
import argparse
from torchvision.transforms import ToTensor
from torchvision.datasets import MNIST
from avalanche.benchmarks import nc_benchmark
from avalanche.training.supervised import EWC
from avalanche.models import MTSimpleMLP
from avalanche.evaluation.metrics import accuracy_metrics
from avalanche.logging import InteractiveLogger
from avalanche.training.plugins import EvaluationPlugin

cuda = 0
model = MTSimpleMLP(hidden_size=128)
optimizer = torch.optim.SGD(model.parameters(), lr=1e-3)
criterion = torch.nn.CrossEntropyLoss()
assert cuda == -1 or cuda >= 0, "cuda must be -1 or >= 0."
device = torch.device(f"cuda:{cuda}" if torch.cuda.is_available() and cuda >= 0 else "cpu")

mnist_train = MNIST(
    root=expanduser("~") + "/.avalanche/data/mnist/",
    train=True,
    download=True,
    transform=ToTensor())
mnist_test = MNIST(
    root=expanduser("~") + "/.avalanche/data/mnist/",
    train=False,
    download=True,
    transform=ToTensor())
scenario = nc_benchmark(mnist_train, mnist_test, 4, task_labels=True, seed=1234,
                        class_ids_from_zero_in_each_exp=True,
                        fixed_class_order=list(range(10)),
                        per_exp_classes={0: 3, 1: 2, 2: 3, 3: 2})
interactive_logger = InteractiveLogger()
eval_plugin = EvaluationPlugin(
    accuracy_metrics(
        epoch=True, experience=True, stream=True),
    loggers=[interactive_logger],
)
strategy = EWC(
    model,
    optimizer,
    criterion,
    0.4,
    mode='online',
    decay_factor=0.3,
    train_epochs=2,
    device=device,
    train_mb_size=128,
    evaluator=eval_plugin,
)

results = []
for experience in scenario.train_stream:
    print("classes in experience: ", experience.classes_in_this_experience)
    strategy.train(experience)
    results.append(strategy.eval(scenario.test_stream))
```

**Comment by AntonioCarta** [comment]
Closing this since it seems to work on the master branch. Feel free to reopen it if it's not.

**Comment by ruchi4295** [comment]
Hello everyone out there.
I am facing an issue of zero accuracy for EWC, LwF, and SI in class incremental scenarios. I am working on a dataset of 6 tasks with 2-3 disjoint classes. I get a task accuracy zero on previous tasks as I encounter the coming tasks. Could anyone help me with the reason, please?
